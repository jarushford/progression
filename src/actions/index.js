export const getTrainingData = (data) => ({
  type: 'GET_TRAINING',
  data
})

export const getProjects = (data) => ({
  type: 'GET_PROJECTS',
  data
})

export const clearProjects = () => ({
  type: 'CLEAR_PROJECTS'
})

export const getAscents = (data) => ({
  type: 'GET_ASCENTS',
  data
})

export const clearAscents = () => ({
  type: 'CLEAR_ASCENTS'
})

export const toggleDiscipline = (discipline) => ({
  type: 'TOGGLE_DISCIPLINE',
  discipline
})

export const setUser = (user) => ({
  type: 'SET_USER',
  user
})

export const logoutUser = () => ({
  type: 'LOGOUT_USER'
})

export const toggleComplete = (key) => ({
  type: 'TOGGLE_COMPLETE',
  key
})
