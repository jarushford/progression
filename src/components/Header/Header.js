import React from 'react'
import { NavLink, Link } from 'react-router-dom'
import '../../main.scss'

export default function App() {
    return (
      <div className="header">
        <NavLink to="/">
          <div className="logo" />
        </NavLink>
        <NavLink to="/training" className="nav-item"> TRAINING </NavLink>
        <NavLink to="/projects" className="nav-item"> PROJECTS </NavLink>
        <NavLink to="/ascents" className="nav-item"> ASCENTS </NavLink>
        <Link to='login'><button>Login / Sign Up</button></Link>
      </div>
    )
}