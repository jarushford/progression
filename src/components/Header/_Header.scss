.header {
  display: flex;
  align-items: center;
  background: $darkGrey;
  opacity: .94;
  padding-left: 107px;
}

.logo {
  width: 144px;
  height: 107px;
  background: url('./assets/progressionLogo.png');
  background-size: cover;
  position: absolute;
  top: -36px;
  left: -36px;
}

.nav-item {
  color: $text;
  text-decoration: none;
  font-size: 2rem;
  height: 88px;
  line-height: 280%;
  width: 200px;
  text-align: center;
  transition: .3s ease;

  &:hover {
    background: $lightGrey;
    color: $darkGrey;
  }
}

.active,
.active:hover {
  color: $accent;
}