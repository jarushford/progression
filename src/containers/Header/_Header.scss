.header {
  display: flex;
  align-items: center;
  background: $darkGrey;
  opacity: .97;
  padding-left: 115px;
  position: fixed;
  z-index: 2;
  width: 100%;
  box-shadow: 0 1px 30px 1px rgba(200, 200, 200, 0.1);

  button {
    @include button();
    position: absolute;
    top: 25px;
    right: 20px;
  }
}

.logo {
  width: 144px;
  height: 107px;
  background: url('./assets/progressionLogo.png');
  background-size: cover;
  position: absolute;
  top: -36px;
  left: -36px;
}

.nav-item {
  color: $text;
  text-decoration: none;
  font-size: 2rem;
  height: 88px;
  line-height: 280%;
  width: 200px;
  text-align: center;
  transition: .3s ease;

  &:hover {
    background: $lightGrey;
    color: $darkGrey;
  }
}

.active,
.active:hover {
  color: $accent;
}

.current-user {
  position: absolute;
  top: 0;
  right: 15px;
  display: flex;

  .fa-user-circle {
    font-size: 2.7rem;
    color: $accent;
    margin-top: 20px;
  }

  div {
    margin-top: 20px;
    transition: .3s ease;
    border-radius: 6px;
    border-top-right-radius: 0;
  }
  
  h2 {
    margin: 3px 3px 0;
    font-size: 1.2rem;
    color: $text;
    text-align: center;
    transition: .3s ease;
    padding: 10px;
    cursor: pointer;
  }

  h2:nth-child(1) {
    font-weight: 400;
  }

  div:hover {
    background: $midGrey;
  }

  .fa-user-circle {
    height: 47px;
    padding: 2px;
    border-top-right-radius: 10px;
    border-bottom-right-radius: 10px;
    transition: .3s ease;
  }

  .fa-caret-down {
    transition: .4s ease;
    padding-right: 2px;
  }

  div:hover .fa-caret-down {
    transform: rotate(180deg);
    padding-bottom: 2px;
    padding-right: 2px;
  }

  div:hover ~ .fa-user-circle {
    background: $midGrey;
  }

  h2:nth-child(1):hover ~ .logout-link h2 {
    opacity: 1;
    height: 50px;
  }

  .logout-link h2 {
    margin-top: 0;
    background: $midGrey;
    height: 0px;
    padding-top: 10px;
    opacity: 0;
    transition: .3s ease;
    
    &:hover {
      height: 50px;
      opacity: 1;
      color: $accent;
    }
  }
}

.logout-link {
  text-decoration: none;
}