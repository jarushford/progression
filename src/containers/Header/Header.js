import React from 'react'
import { NavLink, Link, withRouter } from 'react-router-dom'
import '../../main.scss'
import { connect } from 'react-redux';
import { logoutUser, clearAscents, clearProjects, clearWorkouts } from '../../actions'

function Header({ currentUser, logoutUser, clearAscents, clearProjects, clearWorkouts }) {
  let userButton

  if (currentUser.name) {
    userButton = (
      <div className="current-user">
        <div>
          <h2><i className="fas fa-caret-down" /> Welcome, {currentUser.name}!</h2>
          <Link
            to="/"
            className="logout-link"
            onClick={() => {
              clearAscents()
              clearProjects()
              clearWorkouts()
              logoutUser()
            }}
          >
            <h2>Log Out</h2>
          </Link>
        </div>
        <i className="fas fa-user-circle" />
      </div>
    )
  } else {
    userButton = (
      <Link to='/login'><button>Login / Sign Up</button></Link>
    )
  }

    return (
      <div className="header">
        <NavLink to="/">
          <div className="logo" />
        </NavLink>
        <NavLink to="/training" className="nav-item"> TRAINING </NavLink>
        <NavLink to="/projects" className="nav-item"> PROJECTS </NavLink>
        <NavLink to="/ascents" className="nav-item"> ASCENTS </NavLink>
        {userButton}
      </div>
    )
}

const mapStateToProps = (state) => ({
  currentUser: state.currentUser
})

const mapDispatchToProps = (dispatch) => ({
  clearAscents: () => dispatch(clearAscents()),
  clearProjects: () => dispatch(clearProjects()),
  clearWorkouts: () => dispatch(clearWorkouts()),
  logoutUser: () => dispatch(logoutUser())
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Header))