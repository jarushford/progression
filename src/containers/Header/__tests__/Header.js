import React from 'react'
import { shallow } from 'enzyme'
import { logoutUser, clearAscents, clearProjects, clearWorkouts } from '../../../actions'
import { mapStateToProps, mapDispatchToProps, Header } from '../Header'

jest.mock('../../../actions')

describe('Header', () => {
  let mockUser
  let mockClearAscents
  let mockClearProjects
  let mockClearWorkouts
  let mockLogoutUser
  let mockToggle

  beforeEach(() => {
    mockUser = {
      name: 'Frodo',
      email: 'thechosenone@hobbiton.org',
      password: 'Sam'
    }
    mockClearAscents = jest.fn()
    mockClearProjects = jest.fn()
    mockClearWorkouts = jest.fn()
    mockLogoutUser = jest.fn()
    mockToggle = jest.fn()
  })

  describe('Header Component', () => {
    it('should match the snapshot', () => {
      const wrapper = shallow(<Header currentUser={mockUser} clearAscents={mockClearAscents} clearProjects={mockClearProjects} clearWorkouts={mockClearWorkouts} logoutUser={mockLogoutUser} toggleMenu={mockToggle} />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot with no user', () => {
      const noUser = { name: '', email: '', password: '' }
      const wrapper = shallow(<Header currentUser={noUser} clearAscents={mockClearAscents} clearProjects={mockClearProjects} clearWorkouts={mockClearWorkouts} logoutUser={mockLogoutUser} toggleMenu={mockToggle} />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should logout a user when they click on log out', () => {
      const wrapper = shallow(<Header currentUser={mockUser} clearAscents={mockClearAscents} clearProjects={mockClearProjects} clearWorkouts={mockClearWorkouts} logoutUser={mockLogoutUser} toggleMenu={mockToggle} />)

      wrapper.find('.logout-link').first().simulate('click')

      expect(mockClearAscents).toBeCalled()
      expect(mockClearProjects).toBeCalled()
      expect(mockClearWorkouts).toBeCalled()
      expect(mockLogoutUser).toBeCalled()
    })
  })

  describe('mapStateToProps', () => {
    const mockState = {
      currentUser: mockUser
    }

    const mappedProps = mapStateToProps(mockState)

    expect(mappedProps).toEqual(mockState)
  })

  describe('mapDispatchToProps', () => {
    let mockDispatch
    let mappedProps

    beforeEach(() => {
      mockDispatch = jest.fn()
      mappedProps = mapDispatchToProps(mockDispatch)
    })

    it('should return a props object with a method clearAscents', () => {
      const expected = clearAscents()

      mappedProps.clearAscents()

      expect(mockDispatch).toBeCalledWith(expected)
    })

    it('should return a props object with a method clearProjects', () => {
      const expected = clearProjects()

      mappedProps.clearProjects()

      expect(mockDispatch).toBeCalledWith(expected)
    })

    it('should return a props object with a method clearWorkouts', () => {
      const expected = clearWorkouts()

      mappedProps.clearWorkouts()

      expect(mockDispatch).toBeCalledWith(expected)
    })

    it('should return a props object with a method logoutUser', () => {
      const expected = logoutUser()

      mappedProps.logoutUser()

      expect(mockDispatch).toBeCalledWith(expected)
    })
  })
})