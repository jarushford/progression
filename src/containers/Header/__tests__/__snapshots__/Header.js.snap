// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    currentUser={
      Object {
        "email": "thechosenone@hobbiton.org",
        "name": "Frodo",
        "password": "Sam",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/"
        >
          <div
            className="logo"
          />
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-item"
          to="/training"
        >
           TRAINING 
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-item"
          to="/projects"
        >
           PROJECTS 
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-item"
          to="/ascents"
        >
           ASCENTS 
        </NavLink>,
        <div
          className="current-user"
        >
          <div>
            <h2>
              <i
                className="fas fa-caret-down"
              />
               Welcome, 
              Frodo
              !
            </h2>
            <Link
              className="logout-link"
              onClick={[Function]}
              replace={false}
              to="/"
            >
              <h2>
                Log Out
              </h2>
            </Link>
          </div>
          <i
            className="fas fa-user-circle"
          />
        </div>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <div
            className="logo"
          />,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "logo",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": " TRAINING ",
          "className": "nav-item",
          "to": "/training",
        },
        "ref": null,
        "rendered": " TRAINING ",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": " PROJECTS ",
          "className": "nav-item",
          "to": "/projects",
        },
        "ref": null,
        "rendered": " PROJECTS ",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": " ASCENTS ",
          "className": "nav-item",
          "to": "/ascents",
        },
        "ref": null,
        "rendered": " ASCENTS ",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <h2>
                <i
                  className="fas fa-caret-down"
                />
                 Welcome, 
                Frodo
                !
              </h2>
              <Link
                className="logout-link"
                onClick={[Function]}
                replace={false}
                to="/"
              >
                <h2>
                  Log Out
                </h2>
              </Link>
            </div>,
            <i
              className="fas fa-user-circle"
            />,
          ],
          "className": "current-user",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  <i
                    className="fas fa-caret-down"
                  />
                   Welcome, 
                  Frodo
                  !
                </h2>,
                <Link
                  className="logout-link"
                  onClick={[Function]}
                  replace={false}
                  to="/"
                >
                  <h2>
                    Log Out
                  </h2>
                </Link>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      className="fas fa-caret-down"
                    />,
                    " Welcome, ",
                    "Frodo",
                    "!",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-caret-down",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  " Welcome, ",
                  "Frodo",
                  "!",
                ],
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <h2>
                    Log Out
                  </h2>,
                  "className": "logout-link",
                  "onClick": [Function],
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Log Out",
                  },
                  "ref": null,
                  "rendered": "Log Out",
                  "type": "h2",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-user-circle",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/"
          >
            <div
              className="logo"
            />
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav-item"
            to="/training"
          >
             TRAINING 
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav-item"
            to="/projects"
          >
             PROJECTS 
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav-item"
            to="/ascents"
          >
             ASCENTS 
          </NavLink>,
          <div
            className="current-user"
          >
            <div>
              <h2>
                <i
                  className="fas fa-caret-down"
                />
                 Welcome, 
                Frodo
                !
              </h2>
              <Link
                className="logout-link"
                onClick={[Function]}
                replace={false}
                to="/"
              >
                <h2>
                  Log Out
                </h2>
              </Link>
            </div>
            <i
              className="fas fa-user-circle"
            />
          </div>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <div
              className="logo"
            />,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "logo",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": " TRAINING ",
            "className": "nav-item",
            "to": "/training",
          },
          "ref": null,
          "rendered": " TRAINING ",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": " PROJECTS ",
            "className": "nav-item",
            "to": "/projects",
          },
          "ref": null,
          "rendered": " PROJECTS ",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": " ASCENTS ",
            "className": "nav-item",
            "to": "/ascents",
          },
          "ref": null,
          "rendered": " ASCENTS ",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <h2>
                  <i
                    className="fas fa-caret-down"
                  />
                   Welcome, 
                  Frodo
                  !
                </h2>
                <Link
                  className="logout-link"
                  onClick={[Function]}
                  replace={false}
                  to="/"
                >
                  <h2>
                    Log Out
                  </h2>
                </Link>
              </div>,
              <i
                className="fas fa-user-circle"
              />,
            ],
            "className": "current-user",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2>
                    <i
                      className="fas fa-caret-down"
                    />
                     Welcome, 
                    Frodo
                    !
                  </h2>,
                  <Link
                    className="logout-link"
                    onClick={[Function]}
                    replace={false}
                    to="/"
                  >
                    <h2>
                      Log Out
                    </h2>
                  </Link>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <i
                        className="fas fa-caret-down"
                      />,
                      " Welcome, ",
                      "Frodo",
                      "!",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "fas fa-caret-down",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    " Welcome, ",
                    "Frodo",
                    "!",
                  ],
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <h2>
                      Log Out
                    </h2>,
                    "className": "logout-link",
                    "onClick": [Function],
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Log Out",
                    },
                    "ref": null,
                    "rendered": "Log Out",
                    "type": "h2",
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-user-circle",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Header Header Component should match the snapshot with no user 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    currentUser={
      Object {
        "email": "",
        "name": "",
        "password": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/"
        >
          <div
            className="logo"
          />
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-item"
          to="/training"
        >
           TRAINING 
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-item"
          to="/projects"
        >
           PROJECTS 
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-item"
          to="/ascents"
        >
           ASCENTS 
        </NavLink>,
        <Link
          replace={false}
          to="/login"
        >
          <button>
            Login / Sign Up
          </button>
        </Link>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <div
            className="logo"
          />,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "logo",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": " TRAINING ",
          "className": "nav-item",
          "to": "/training",
        },
        "ref": null,
        "rendered": " TRAINING ",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": " PROJECTS ",
          "className": "nav-item",
          "to": "/projects",
        },
        "ref": null,
        "rendered": " PROJECTS ",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": " ASCENTS ",
          "className": "nav-item",
          "to": "/ascents",
        },
        "ref": null,
        "rendered": " ASCENTS ",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button>
            Login / Sign Up
          </button>,
          "replace": false,
          "to": "/login",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login / Sign Up",
          },
          "ref": null,
          "rendered": "Login / Sign Up",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/"
          >
            <div
              className="logo"
            />
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav-item"
            to="/training"
          >
             TRAINING 
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav-item"
            to="/projects"
          >
             PROJECTS 
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav-item"
            to="/ascents"
          >
             ASCENTS 
          </NavLink>,
          <Link
            replace={false}
            to="/login"
          >
            <button>
              Login / Sign Up
            </button>
          </Link>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <div
              className="logo"
            />,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "logo",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": " TRAINING ",
            "className": "nav-item",
            "to": "/training",
          },
          "ref": null,
          "rendered": " TRAINING ",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": " PROJECTS ",
            "className": "nav-item",
            "to": "/projects",
          },
          "ref": null,
          "rendered": " PROJECTS ",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": " ASCENTS ",
            "className": "nav-item",
            "to": "/ascents",
          },
          "ref": null,
          "rendered": " ASCENTS ",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button>
              Login / Sign Up
            </button>,
            "replace": false,
            "to": "/login",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login / Sign Up",
            },
            "ref": null,
            "rendered": "Login / Sign Up",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
